src/bp/ui-shared/src/TreeView/index.tsx

```diff
@@ -53,14 +53,26 @@ const TreeView = <T extends {}>(props: TreeViewProps<T>) => {
     setExpanded(props.expandedPaths || [])
   }, [props.expandedPaths])
 
-  const handleNodeExpansion = (node: TreeNode<T>, isExpanded: boolean) => {
+  const handleInitialNodeExpansion = (node: TreeNode<T>, isExpanded: boolean) => {
+    changeNodeExpansion(node, isExpanded)
+
+    if (node.parent) {
+      handleInitialNodeExpansion(node.parent!, isExpanded)
+    }
+  }
+
+  const changeNodeExpansion = (node: TreeNode<T>, isExpanded: boolean) => {
     isExpanded ? setExpanded([...expanded, node.fullPath]) : setExpanded(expanded.filter(x => x !== node.fullPath))
     node.isExpanded = isExpanded
 
     props.onExpandToggle?.(node, isExpanded)
   }
 
   const handleNodeClick = (selectedNode: TreeNode<T>) => {
+    if (selectedNode?.nodeData?.type === 'qna') {
+      return
+    }
+
```

Comment:
Please avoid custom logic in ui-shared. The goal is to make it as generic as possible, then implement them where it's being used. It will replace the tree view in flows & in the code editor in the future.

Reply:
I can't because it will still select the node and that's what I'm trying to avoid...

Reply:
Added isNotClickable to NodeData, that way it's still generic, but in TreeView